.github/actions/build/action.yml

```diff
@@ -0,0 +1,44 @@
+name: Build
+description: 'Build action to build and validate the MonoGame Framework'
+inputs:
+  platform:
+    description: 'Platform to run the action on'
+    required: true
+  shell:
+    description: 'Shell to use for the action'
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: Build
+      run: dotnet run --project build/Build.csproj -- --target=Default
+      shell: ${{ inputs.shell }}
+
+    - name: Run Tests Windows

```

Comment:
Instead of making "Run Tests Windows" and "Run Tests Linux/Mac", please make a "Run Tests" that in the shell execution checks the system its running on and executes appropriate commands.